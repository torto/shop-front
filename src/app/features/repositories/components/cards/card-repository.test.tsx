import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CardRepository from "./card-repository";

const setup = (props: any) => {
    const newProps = {
      name: "name",
      html_url: "http://aa.com",
      description: "description",
      created_at: "2020-10-04T12:04:29.676Z",
      avatar_url: "http://aa.com.image.jps",
      stargazers_count: 1000,
      type: "User",
      login: "torto",
      isStar: false,
      handleToggleStar: jest.fn(),
      ...props
    };
    
    const target = render(<CardRepository {...newProps} />);
    return target;

}


test("renders card - isStart: false - avatar is present", () => {
  const { asFragment } = setup({});

  expect(asFragment()).toMatchSnapshot();
});
test("renders card - isStart: true - avatar is not present", () => {
  const { asFragment } = setup({
    isStar: true,
    avatar_url: ''
  });

  expect(asFragment()).toMatchSnapshot();
});

test("trigger start click", () => {
    const query = {
      handleToggleStar: jest.fn(),
    };
  const { getByRole } = setup(query);
  fireEvent.click(getByRole('button'));
  expect(query.handleToggleStar).toBeCalled();
});
