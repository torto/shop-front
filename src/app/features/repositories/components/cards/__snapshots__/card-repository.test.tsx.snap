// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card - isStart: false - avatar is present 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root CardRepository-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="recipe"
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            class="MuiAvatar-img"
            src="http://aa.com.image.jps"
          />
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          <strong>
            User - torto
          </strong>
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Created - Sun, 04 Oct 2020 12:04:29 GMT
        </span>
      </div>
    </div>
    <div
      class="MuiCardMedia-root CardRepository-media-2"
      style="background-image: url(github.png);"
      title="Paella dish"
    />
    <div
      class="MuiCardContent-root"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h5 MuiTypography-colorPrimary"
        href="http://aa.com"
        target="_blank"
      >
        name
      </a>
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        description
      </p>
    </div>
    <div
      class="MuiCardActions-root"
    >
      <button
        aria-label="add to favorites"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root CardRepository-numberStar-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        1000
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders card - isStart: true - avatar is not present 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root CardRepository-card-5 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="recipe"
          class="MuiAvatar-root MuiAvatar-circle CardRepository-avatar-7 MuiAvatar-colorDefault"
        >
          N
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          <strong>
            User - torto
          </strong>
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Created - Sun, 04 Oct 2020 12:04:29 GMT
        </span>
      </div>
    </div>
    <div
      class="MuiCardMedia-root CardRepository-media-6"
      style="background-image: url(github.png);"
      title="Paella dish"
    />
    <div
      class="MuiCardContent-root"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h5 MuiTypography-colorPrimary"
        href="http://aa.com"
        target="_blank"
      >
        name
      </a>
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        description
      </p>
    </div>
    <div
      class="MuiCardActions-root"
    >
      <button
        aria-label="add to favorites"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root CardRepository-numberStar-8 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        1000
      </p>
    </div>
  </div>
</DocumentFragment>
`;
